src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -161,6 +190,7 @@ private void GenModelSelfValidationIfNecessary(ValidatedModel modelToValidate)
         {
             if (modelToValidate.SelfValidates)
             {
+                OutLn($"builder ??= new global::Microsoft.Extensions.Options.ValidateOptionsResultBuilder();");
                 OutLn($"builder.AddResults(((global::System.ComponentModel.DataAnnotations.IValidatableObject)options).Validate(context));");
```

Comment:
What you have is fine. It could also be:
```C#
OutLn($"(builder ??= new()).AddResults(((global::System.ComponentModel.DataAnnotations.IValidatableObject)options).Validate(context));");
```